informer/src/informer_service/dummy_informer_service.rs

```diff
@@ -0,0 +1,96 @@
+// Copyright 2020 Kodebox, Inc.
+// This file is part of CodeChain.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+use crate::{informer_notify, Connection, NotifySender};
+use crossbeam::Receiver;
+use crossbeam_channel as crossbeam;
+use std::thread::spawn;
+
+#[derive(Clone, PartialEq)]
+pub enum Events {
+    None,
+    PeerAdded(String, String, usize),
+}
+#[derive(Clone)]
+pub struct InformerService {
+    connections: Vec<Connection>,
+    pub sender: NotifySender,
```

Comment:
Why does the InformerService have a sender? I thought that the InformerService would have only receivers.

Reply:
Because I create a channel for sending connection in  `run node` and share with `handler` and `informer service`. There

Reply:
IMO, since the sender is not used in the InformerService, it should not have it.
There are two options for how to remove the sender from the InfomreService struct.

One: making InformerService::new return the sender. For example
```rs
pub fn new(connection_receiver: Receiver<Connection>) -> (Self, NotifySender) {
```

Two: making the InformerService::new receive the sender. For example:
```rs
    pub fn new(connection_receiver: Receiver<Connection>, event_receiver: Receiver<Events>) -> Self {
```

Reply:
I got your point and am agree with you. I will use the first way.